### ISSUE TITLE
MySQL_DB Module collation is not respected.

### ISSUE THREAD

**Issue by ghost** [ansible/ansible-modules-core#1317]
I have created several mysql databases via Ansible using following script:

``` yaml
- name: Create MySQL Database
  mysql_db: >
    name=mydb
    collation=utf8_swedish_ci
    encoding=utf8
    state=present
```

but the database ends up with **utf8_general_ci** collation.

``` shell
*************************** 478. row ***************************
           Name: wysiwyg_user
         Engine: InnoDB
        Version: 10
     Row_format: Compact
           Rows: 0
 Avg_row_length: 0
    Data_length: 16384
Max_data_length: 0
   Index_length: 32768
      Data_free: 152043520
 Auto_increment: NULL
    Create_time: 2015-05-04 12:46:41
    Update_time: NULL
     Check_time: NULL
      Collation: utf8_general_ci
       Checksum: NULL
 Create_options: 
        Comment: Stores user preferences for wysiwyg profiles.

```

**Comment by blackreed9** [comment]
what version of Ansible and mysql are  you using?

I'm not able to reproduce this on my CentOS 6.3 system.

#### Playbook

```
---
-
  gather_facts: false
  hosts: test
  tasks:
    -
      name: Testing PR 1317 - Create Swedish DB
      mysql_db: >
        name=mydb
        collation=utf8_swedish_ci
        encoding=utf8
        state=present
```

#### Result

```
mysql> select * from information_schema.schemata where schema_name = 'mydb';
+--------------+-------------+----------------------------+------------------------+----------+
| CATALOG_NAME | SCHEMA_NAME | DEFAULT_CHARACTER_SET_NAME | DEFAULT_COLLATION_NAME | SQL_PATH |
+--------------+-------------+----------------------------+------------------------+----------+
| NULL         | mydb        | utf8                       | utf8_swedish_ci        | NULL     |
+--------------+-------------+----------------------------+------------------------+----------+
1 row in set (0.00 sec)
```

#### MySQL

```
Name        : mysql-server
Arch        : x86_64
Version     : 5.1.73
Release     : 3.el6_5
```

**Comment by ghost** [comment]
```
ansible --version
ansible 1.9.1

mysql --version
mysql  Ver 15.1 Distrib 5.5.41-MariaDB, for Linux (x86_64) using readline 5.1

CentOS 7
```

**Comment by Jmainguy** [comment]
Unable to reproduce using ansible 2.0 on centos7 with mariadb

```
[root@phy01 ansible]# ansible localhost -m ping
localhost | success >> {
    "changed": false,
    "ping": "pong"
}

[root@phy01 ansible]# ansible localhost -m mysql_db -a "name=jon collation=utf8_swedish_ci encoding=utf8 state=present"
localhost | success >> {
    "changed": true,
    "db": "jon"
}

[root@phy01 ansible]# ansible localhost -m mysql_db -a "name=jon collation=utf8_swedish_ci encoding=utf8 state=present"
localhost | success >> {
    "changed": false,
    "db": "jon"
}

[root@phy01 ansible]# mysql
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 1153407
Server version: 5.5.41-MariaDB MariaDB Server

Copyright (c) 2000, 2014, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]>  select * from information_schema.schemata where schema_name = 'jon';                                                                         
+--------------+-------------+----------------------------+------------------------+----------+
| CATALOG_NAME | SCHEMA_NAME | DEFAULT_CHARACTER_SET_NAME | DEFAULT_COLLATION_NAME | SQL_PATH |
+--------------+-------------+----------------------------+------------------------+----------+
| def          | jon         | utf8                       | utf8_swedish_ci        | NULL     |
+--------------+-------------+----------------------------+------------------------+----------+
1 row in set (0.00 sec)
```

**Comment by bcoca** [comment]
I cannot reproduce this, but I did see that collation can only be set on creation, so if I change it after the database is created, it does not update the db. To do so it would need to drop and recreate, which I don't think anyone wants.

**Comment by Jmainguy** [comment]
@rajibmp any opinion on bcoca's comment above? Should we keep this open?

**Comment by ghost** [comment]
@Jmainguy I agree with @bcoca that its not feasible to drop and recreate database. 

At the moment the issue is not with the database (schema) collation itself. Now when I create the database with the original script I posted, the schema collation is `utf8_swedish_ci` but when I create a table in the database without specifying the collation then the table is created as `utf8_general_ci`.  

IMO, if the database has one collation and table is created without explicitly assigning the collation then it should take the collation of database as default. But this might be issue with version of MariaDB I'm using. I'll inform the MariaDB team about it.

**Comment by Jmainguy** [comment]
@rajibmp yeah that seems like a good default.

**Comment by bcoca** [comment]
fixed via 89957eed537b08001ea171ba6a4ead41d31ab983
